stamen/map-compare
# Fix polling url issue

Closes https://github.com/stamen/map-compare/issues/62

This fixes a bug introduced by the work allowing more than 2 maps where switching between more than one locally hosted style caused flashing between the two.

This was caused by polling being called on unexpected remounts (due to a key property we introduced for multiple map work) and not stopping polling on destroying a map.

### QA

Confirm polling works as expected:
- [ ] Enter a custom url to a locally served map style
- [ ] Change the stylesheet and expect to see a change (or console log polling)
- [ ] Change url to another locally served map style
- [ ] Expect the switch to happen with no flashing between maps
- [ ] Change the stylesheet to confirm live changes are being polled

Confirm no regressions in map style input dropdown
- [ ] Compare three maps with different names
- [ ] Remove the first map
- [ ] Expect all dropdowns for maps to be correct
- [ ] Compare more maps, adding and removing, trying to break the dropdown (to show the wrong map style name) (you should not be able to)
